### PR TITLE
fix(gomod): Fix gomod digest updates

### LINKED ISSUE #12645: Go modules digest update is not working
### How are you running Renovate?

WhiteSource Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Describe the bug

Existing Go modules update PRs referring commit digest are closed on Oct 27 and no longer created.

Reproduction: https://github.com/at-wat/renovate-repro-gomod-digest/pull/1
Current version of `golang.org/x/net` is `v0.0.0-20210825183410-e898025ed96a` and latest version is `v0.0.0-20211112202133-69e39bad7dc2` (https://github.com/golang/net/commit/69e39bad7dc2bbb411fa35755c46020969029fa7), but treated as no update.

On my private sandbox repo, unmerged PR seems be closed at `2021-10-27T17:48:50Z` (18 days ago).
image

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: Matched 1 file(s) for manager gomod: go.mod
DEBUG: Found gomod package files
DEBUG: Found 1 package file(s)
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {
      "gomod": {
        "fileCount": 1,
        "depCount": 3
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 3
    }
  }
}
DEBUG: Found no results from datasource that look like a version (golang.org/x/net)(dependency="golang.org/x/net")
{
  "result": {
    "sourceUrl": "https://github.com/golang/net",
    "releases": []
  }
}
DEBUG: Found no results from datasource that look like a version (github.com/seqsense/pcgol)(dependency="github.com/seqsense/pcgol")
{
  "result": {
    "sourceUrl": "https://github.com/seqsense/pcgol",
    "releases": []
  }
}
DEBUG: Found no results from datasource that look like a version (github.com/at-wat/renovate-repro-gomod-dummy)(dependency="github.com/at-wat/renovate-repro-gomod-dummy")
{
  "result": {
    "sourceUrl": "https://github.com/at-wat/renovate-repro-gomod-dummy",
    "releases": []
  }
}
DEBUG: Package releases lookups complete
{
  "baseBranch": "master"
}
DEBUG: branchifyUpgrades
DEBUG: 0 flattened updates found:
DEBUG: Returning 0 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "config": {
    "gomod": [
      {
        "packageFile": "go.mod",
        "constraints": {
          "go": "^1.17"
        },
        "deps": [
          {
            "managerData": {
              "lineNumber": 4
            },
            "depName": "golang.org/x/net",
            "depType": "require",
            "currentValue": "v0.0.0-20210825183410-e898025ed96a",
            "datasource": "go",
            "currentDigest": "e898025ed96a",
            "digestOneAndOnly": true,
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "versioning": "semver",
            "sourceUrl": "https://github.com/golang/net",
            "currentVersion": "v0.0.0-20210825183410-e898025ed96a",
            "fixedVersion": "v0.0.0-20210825183410-e898025ed96a"
          },
          {
            "managerData": {
              "lineNumber": 7,
              "multiLine": true
            },
            "depName": "github.com/at-wat/renovate-repro-gomod-dummy",
            "depType": "require",
            "currentValue": "v0.0.0-20211025070218-857fe7d90bf1",
            "datasource": "go",
            "currentDigest": "857fe7d90bf1",
            "digestOneAndOnly": true,
            "depIndex": 1,
            "updates": [],
            "warnings": [],
            "versioning": "semver",
            "sourceUrl": "https://github.com/at-wat/renovate-repro-gomod-dummy",
            "currentVersion": "v0.0.0-20211025070218-857fe7d90bf1",
            "fixedVersion": "v0.0.0-20211025070218-857fe7d90bf1"
          },
          {
            "managerData": {
              "lineNumber": 8,
              "multiLine": true
            },
            "depName": "github.com/seqsense/pcgol",
            "depType": "require",
            "currentValue": "v0.0.0-20211020051623-80c7a95f592a",
            "datasource": "go",
            "currentDigest": "80c7a95f592a",
            "digestOneAndOnly": true,
            "depIndex": 2,
            "updates": [],
            "warnings": [],
            "versioning": "semver",
            "sourceUrl": "https://github.com/seqsense/pcgol",
            "currentVersion": "v0.0.0-20211020051623-80c7a95f592a",
            "fixedVersion": "v0.0.0-20211020051623-80c7a95f592a"
          }
        ]
      }
    ]
  }
}
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

🤦‍♂️

## Context:

Closes #12645

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/go/index.ts [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { getDigest as _getDigest } from './digest';
 import * as direct from './releases-direct';
 import * as goproxy from './releases-goproxy';
 
 export { id } from './common';
```
Comment (viceice) [inline lib/datasource/go/index.ts:6]: Please use this export style

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/datasource/go/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/datasource/go/index.ts:6]: I've tried, but it caused coverage to fail with 100% value being reported. This explicit import/export is the first thing I tried, but maybe we have a better solution?

Comment (viceice) [reply lib/datasource/go/index.ts:6]: Yes, you need to import it from this file in tests to get it covered. 😉

Comment (zharinov) [reply lib/datasource/go/index.ts:5]: Thanks!

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**